### PR TITLE
adds literaly base64 support, adds importing from a keystore not just…

### PR DESCRIPTION
**Pull request by maxres-fr**
… keypair

### CODE REVIEW

File: config/samples/secret-agent_v1alpha1_secretagentconfiguration.yaml [hunk 1]
```diff
@@ -85,39 +112,72 @@ spec:
...
+        length: 20
+    - name: "AM_PASSWORDS_AMADMIN_CLEAR"
+      type: password
+      spec:
+        length: 24
+    - name: "AM_SELFSERVICE_LEGACY_CONFIRMATION_EMAIL_LINK_SIGNING_KEY"
+      type: password
+      spec:
+        length: 24
+    - name: "AM_SESSION_STATELESS_ENCRYPTION_KEY"
```
Comment (wstrange) [inline config/samples/secret-agent_v1alpha1_secretagentconfiguration.yaml:159]: base64 encode this

File: config/samples/secret-agent_v1alpha1_secretagentconfiguration.yaml [hunk 2]
```diff
@@ -85,39 +112,72 @@ spec:
...
+      type: password
+      spec:
+        length: 24
+    - name: "AM_SESSION_STATELESS_SIGNING_KEY"
+      type: password
```
Comment (wstrange) [inline config/samples/secret-agent_v1alpha1_secretagentconfiguration.yaml:164]: base64 encode this

File: config/samples/secret-agent_v1alpha1_secretagentconfiguration.yaml [hunk 3]
```diff
@@ -85,39 +112,72 @@ spec:
...
+            cmd: importpassword
+            sourcePath: am-passwords/.storepass
+            # sourcePath: "ds-passwords/tmpdsameuserpwd"
+          - name: dsameuserpwd
+            cmd: importpassword
+            sourcePath: "ds-passwords/dirmanager.pw"
+  # AM secrets
+  - name: am-env-secrets
+    keys:
+    - name: "AM_AUTHENTICATION_SHARED_SECRET"
```
Comment (wstrange) [inline config/samples/secret-agent_v1alpha1_secretagentconfiguration.yaml:135]: base64 encode this. See 
https://stash.forgerock.org/users/warren.strange_forgerock.com/repos/forgeops/browse/docker/7.0/am/custom-entrypoint.sh 
We need to base64 a few of the secrets

File: pkg/generator/keytool.go [hunk 4]
```diff
@@ -90,8 +90,8 @@ func (kt *KeyTool) baseCommand(execCmd string) func(cmdName string, args []strin
 		case "-importkeystore":
 			baseArgs = []string{
 				"-deststoretype", string(kt.V1Spec.StoreType),
-				"-deststorepass", kt.keyPassValue,
```
Comment (jrcast) [inline pkg/generator/keytool.go:93]: weird. I am pretty sure I had fixed this before 🤷 merge miss somewhere?

**Pull request merged**